### PR TITLE
Add CI check whether code is correctly formatted

### PR DESCRIPTION
**Pull request by bernhardmgruber**
I have this one running in LLAMA for a few days now and found it useful. Let's give it a try!

Depends on:

- [x] #1218

### CODE REVIEW

**Comment by psychocoderHPC** [comment]
note: If you like to test commits there is no need to open a pull request. If you have not disabled github actions in your fork each commit in the fork will automatically checked. 
I mostly disable all checks except the check I like to test, try to finish the feature and that open the PR. This keeps more CI resources free.

You clang format detected a formating issue in alpaka +1. I think it is because the HEADER reordering #1203 is not included.

**Comment by bernhardmgruber** [comment]
Thanks for the tips! But this is not a test, I would like to merge this into alpaka ;)

> I mostly disable all checks except the check I like to test, try to finish the feature and that open the PR. This keeps more CI resources free.

How do you disable tests? By editing the ci.yml file or is there a GUI option on GitHub?

**Comment by BenjaminW3** [comment]
We may have to reformat the code before a rebased version of this PR succeeds.

File: .github/workflows/ci.yml [hunk 1]
```diff
@@ -72,6 +72,14 @@ env:
   ALPAKA_ACC_GPU_HIP_ONLY_MODE: OFF
 
 jobs:
+  clang-format:
+    runs-on: ubuntu-latest
+    steps:
+    - uses: actions/checkout@v2
+    - uses: DoozyX/clang-format-lint-action@v0.11
+      with:
+        exclude: './thirdParty'
```
Comment (BenjaminW3) [inline .github/workflows/ci.yml]: We should not need this exclude anymore after #781 has been merged.

Comment (bernhardmgruber) [reply .github/workflows/ci.yml]: Right!

**Review by BenjaminW3 — DISMISSED** [review]

**Review by psychocoderHPC — APPROVED** [review]

**Pull request merged**